### PR TITLE
Remove SHA1 FIPS selftest now we have removed SHA1 from allowed FIPS algorithms.

### PR DESCRIPTION
**Pull request by jallisonciq**
Jason's patch.

### CODE REVIEW

File: crypto/fips/fips_hmac_selftest.c [hunk 1]
```diff
@@ -62,14 +62,6 @@ typedef struct {
...
 
 static const HMAC_KAT vector[] = {
-    {EVP_sha1,
-     /* from http://csrc.nist.gov/publications/fips/fips198/fips-198a.pdf */
-     "0123456789:;<=>?@ABC",
-     "Sample #2",
-     {0x09, 0x22, 0xd3, 0x40, 0x5f, 0xaa, 0x3d, 0x19,
-      0x4f, 0x82, 0xa4, 0x58, 0x30, 0x73, 0x7d, 0x5c,
-      0xc6, 0xc7, 0x5d, 0x24}
-     },
```
Comment (jason-rodri) [inline crypto/fips/fips_hmac_selftest.c:72]: SHA1 is no longer an approved algo we need to remove it from the selftests. This is removing sha1 from the HMAC KAT

File: crypto/fips/fips_post.c [hunk 2]
```diff
@@ -84,8 +84,6 @@ int FIPS_selftest(void)
         ERR_add_error_data(2, "Type=", "rand_drbg_selftest");
         rv = 0;
     }
-    if (!FIPS_selftest_sha1())
-        rv = 0;
```
Comment (jason-rodri) [inline crypto/fips/fips_post.c:88]: remove the sha1 selftest from FIPS post. SHA1 is no longer an approved algo for our FIPS offering.

**Review by solardiz — APPROVED** [review]
Looks reasonable to me. Alternatively, maybe we want to replace those tests with making sure SHA-1 refuses to run? Testing for expected failure is a thing.

**Comment by solardiz** [comment]
> Alternatively, maybe we want to replace those tests with making sure SHA-1 refuses to run? Testing for expected failure is a thing.

This should be part of a bigger, consistent decision, not specific to SHA-1 indeed.

**Comment by jason-rodri** [comment]
> > Alternatively, maybe we want to replace those tests with making sure SHA-1 refuses to run? Testing for expected failure is a thing.
> 
> This should be part of a bigger, consistent decision, not specific to SHA-1 indeed.

I agree that failcases would be a good feature, but the spec doesn't call for it and we need to stick to the spec.

**Pull request merged**